pkg/controller/direct/sql/sqlinstance_controller.go

```diff
@@ -142,7 +142,38 @@ func (a *sqlInstanceAdapter) Create(ctx context.Context, createOp *directbase.Cr
 	if a.desired.Spec.CloneSource != nil {
 		return a.cloneInstance(ctx, u, log)
 	} else {
-		return a.insertInstance(ctx, u, log)
+		// if the maitenance version is provided on CREATE, we need to break up the operation between
+		// a create and a patch with the maintenance version
+		maintenanceVersion := ""
+		if a.desired.Spec.MaintenanceVersion != nil {
+			maintenanceVersion = *a.desired.Spec.MaintenanceVersion
+			a.desired.Spec.MaintenanceVersion = nil
+		}
+
+		if err := a.insertInstance(ctx, u, log); err != nil {
+			return err
```

Comment:
Might be nice to wrap the error to bring some additional context. Eg "failed to perform initial insert: %w"